nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ssh/StandardSSHConfigProvider.java

```diff
@@ -0,0 +1,133 @@
+     *
+     * @param identifier SSH Client identifier for runtime tracking
+     * @param context Property Context
+     * @return SSH Configuration
+     */
+    @Override
+    public Config getConfig(final String identifier, final PropertyContext context) {
+        final DefaultConfig config = new DefaultConfig();
+        final KeepAliveProvider keepAliveProvider = getKeepAliveProvider(identifier, context);
+        config.setKeepAliveProvider(keepAliveProvider);
+
+        getOptionalProperty(context, CIPHERS_ALLOWED).ifPresent(property -> config.setCipherFactories(getFilteredValues(property, config.getCipherFactories())));
+        getOptionalProperty(context, KEY_ALGORITHMS_ALLOWED).ifPresent(property -> config.setKeyAlgorithms(getFilteredValues(property, config.getKeyAlgorithms())));
+        getOptionalProperty(context, KEY_EXCHANGE_ALGORITHMS_ALLOWED).ifPresent(property -> config.setKeyExchangeFactories(getFilteredValues(property, config.getKeyExchangeFactories())));
+        getOptionalProperty(context, MESSAGE_AUTHENTICATION_CODES_ALLOWED).ifPresent(property -> config.setMACFactories(getFilteredValues(property, config.getMACFactories())));
+
+        return config;
+    }
+
+    private Optional<String> getOptionalProperty(final PropertyContext context, final PropertyDescriptor propertyDescriptor) {
+        final PropertyValue propertyValue = context.getProperty(propertyDescriptor);
+        return propertyValue.isSet() ? Optional.of(propertyValue.evaluateAttributeExpressions().getValue()) : Optional.empty();
+    }
+
+    private <T> List<Factory.Named<T>> getFilteredValues(final String propertyValue, final List<Factory.Named<T>> supportedValues) {
+        final Set<String> configuredValues = getCommaSeparatedValues(propertyValue);
+        return supportedValues.stream().filter(named -> configuredValues.contains(named.getName())).collect(Collectors.toList());
+    }
+
+    private Set<String> getCommaSeparatedValues(final String propertyValue) {
+        final String[] values = propertyValue.split(COMMA_SEPARATOR);
+        return Arrays.stream(values).map(String::trim).collect(Collectors.toSet());
+    }
+
+    private KeepAliveProvider getKeepAliveProvider(final String identifier, final PropertyContext context) {
+        final boolean keepAliveEnabled = context.getProperty(USE_KEEPALIVE_ON_TIMEOUT).asBoolean();
+        return keepAliveEnabled ? new EnabledKeepAliveProvider(identifier) : DISABLED_KEEP_ALIVE_PROVIDER;
+    }
+
+    private static class EnabledKeepAliveProvider extends KeepAliveProvider {
+        private final String identifier;
+
+        private EnabledKeepAliveProvider(final String identifier) {
+            this.identifier = identifier;
+        }
+
+        @Override
+        public KeepAlive provide(final ConnectionImpl connection) {
+            final KeepAlive keepAlive = KeepAliveProvider.KEEP_ALIVE.provide(connection);
+            keepAlive.setName(String.format("SSH-keep-alive-%s", identifier));
```

Comment:
Too bad we don't have a way of passing through the processor id.  Although, I guess we won't need it now. :)